presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java

```diff
@@ -75,62 +105,72 @@ public static Response getLogs(String filename, JaxrsParameter<Instant> startDat
+                return badRequest("End of date range can not be before start");
+            }
         }
 
         if (!logLevel.isValid()) {
-            return Response.status(Status.BAD_REQUEST)
-                    .entity("Invalid log level").build();
+            return badRequest("Invalid log level");
         }
         Level level = logLevel.get();
 
-        Path filePath = LOG_DIRECTORY.resolve(filename);
-        if (!Files.isRegularFile(filePath)) {
-            return Response.status(Status.NOT_FOUND)
-                    .entity(Files.exists(filePath)
-                            ? "Not a regular file"
-                            : "File not found")
-                    .build();
+        Path filePath;
+        try {
+            filePath = LOG_DIRECTORY.resolve(filename);
+        }
+        catch (InvalidPathException e) {
+            return badRequest("Invalid file name");
         }
 
         LogFilter logFilter;
         try {
             logFilter = LogFilter.builder()
                     .setFile(filePath)
                     .setPattern(LOG_REGEX)
+                    .setDefaultEntry(DEFAULT_ENTRY)
                     .setLineSeparator("\r\n")
-                    .addGroupFilter(DATE_GROUP, getFilter(start, end))
-                    .addGroupFilter(LEVEL_GROUP, getFilter(level))
                     .setCapacity(maxEntries == null ? Integer.MAX_VALUE : maxEntries)
                     .keepFirst(start != null)
+                    .addGroupFilter(DATE_GROUP, getFilter(start, end))
+                    .addGroupFilter(LEVEL_GROUP, getFilter(level))
                     .build();
         }
         catch (FileNotFoundException e) {
-            // TODO: Log exception, log file disappeared since check for 404
-            return Response.status(Status.INTERNAL_SERVER_ERROR)
-                    .entity("File disappeared while processing operation").build();
+            return Response.status(Status.NOT_FOUND)
+                    .entity(Files.exists(filePath)
+                            ? "Not a regular file"
+                            : "File not found")
+                    .build();
+        }
+        catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+            return internalError("Log parser configured incorrectly",
+                    "Internal: Capturing group not present in log entry pattern", e);
+        }
+        catch (DateTimeParseException e) {
+            return internalError("Log parser configured incorrectly",
+                    "Internal: Default log entry has invalid date", e);
         }
 
         Stream<String> logEntries;
         try {
             logEntries = logFilter.streamEntries();
         }
         catch (IOException e) {
-            return Response.status(Status.INTERNAL_SERVER_ERROR)
-                    .entity("IOException while reading file").build();
+            return internalWarning("IOException while reading file", e);
```

Comment:
You did not catch the `ConcurrentModificationException` thrown by streamEntries() here, ditto for another same method call few lines below.